text/1028-cli-hooks.rst

```diff
@@ -122,23 +125,105 @@ means that the hooks associated with applying schema changes must be executed
 for these commands as well.
 
 
+Watch Configuration
+===================
+
+Sometimes in a Gel project there's a need to respond to some file changes: run
+a migration due to schema change or run code generators due to query file
+change. The CLI already supports watching the schema for changes and applying
+them, but we can generalize this mechanism and make it more flexible, much
+like the hooks that respond to CLI commands.
+
+The idea is to have a mechanism for specifying what changes you want to watch
+and a script that gets triggered by those changes. Unlike hooks, there are no
+"before" and "after" triggers here. Only one kind of response is possible:
+trigger when a change to the watched entity is detected.
+
+The watch configuration will be described in the ``gel.toml`` project manifest
+file in the ``[watch]`` table. We will introduce a ``[watch.files]`` sub-table
+for watching file system changes and triggering scripts. Potentially, we will
+also later introduce a ``[watch.gel-config]`` sub-table for monitoring changes
+to the various database config values and responding to them.
+
+The general structure of watch tables is going to use the key (potentially
+quoted) to specify *what is being watched* and the corresponding value to
+specify the script that will be triggered by changes.
+
+Since the triggered scripts will be executed in response to changes, we cannot
+meaningfully display the output of the scripts to the user. We may want to
+setup an optional logfile to record the script output. This can be specified
+in the general ``[watch]`` section as ``logfile="<path-to-logfile>"``.
```

Comment:
I'm not sure I understand this: why wouldn't you see the output in the same TTY as the running watch command?

Reply:
Fair, but that's commonly solved by actually piping stdout/err, right? I'm fine with a more user-friendly way to configure it like the suggestion here, but I would _hope_ that we'd also see stdout/stderr in the same terminal as watch: I'd be 😱 if I had to tail a log file to see the live output of the scripts.

Reply:
Yes, showing output in TTY is a must. Logfile would just be nice.